stdlib/Random/src/misc.jl

```diff
@@ -71,7 +71,7 @@ function randstring end
 
 let b = UInt8['0':'9';'A':'Z';'a':'z']
     global randstring
-    randstring(r::AbstractRNG, chars=b, n::Integer=8) = String(rand(r, chars, n))
+    randstring(r::AbstractRNG, chars=b, n::Integer=8) = String(rand!(r, Base.StringVector(n), chars))
```

Comment:
It doesn't seem to work in the général case, e.g. with this change:
```julia
julia> randstring("àé", 2)
"\xe0\xe9"
```

Reply:
But this could be made to work in the case where `chars::Vector{UInt8}` at least, or when `isascii(chars)`.

Reply:
The `isascii` test is an interesting thought. That would probably mostly even avoid the performance gotcha mentioned in the docstring.